recipes/jsonnet/all/conandata.yml

```diff
@@ -9,18 +12,37 @@ sources:
     url: "https://github.com/google/jsonnet/archive/v0.17.0.tar.gz"
     sha256: "076b52edf888c01097010ad4299e3b2e7a72b60a41abbc65af364af1ed3c8dbe"
 patches:
+  "0.20.0":
+    - patch_file: "patches/0.18.0/0001-fix-nlohmann-include.patch"
+      patch_description: "fix include path to use cci package"
+      patch_type: "conan"
+    - patch_file: "patches/0.18.0/0002-cmake-fixes.patch"
+      patch_description: "fix rapidyaml name, add installation"
+      patch_type: "conan"
+    - patch_file: "patches/0.20.0/0003-use-cpp17.patch"
+      patch_description: "use C++17"
+      patch_type: "portability"
   "0.19.1":
     - patch_file: "patches/0.18.0/0001-fix-nlohmann-include.patch"
-      base_path: "source_subfolder"
+      patch_description: "fix include path to use cci package"
+      patch_type: "conan"
     - patch_file: "patches/0.18.0/0002-cmake-fixes.patch"
-      base_path: "source_subfolder"
+      patch_description: "fix rapidyaml name, add installation"
+      patch_type: "conan"
   "0.18.0":
     - patch_file: "patches/0.18.0/0001-fix-nlohmann-include.patch"
-      base_path: "source_subfolder"
+      patch_description: "fix include path to use cci package"
+      patch_type: "conan"
     - patch_file: "patches/0.18.0/0002-cmake-fixes.patch"
-      base_path: "source_subfolder"
+      patch_description: "fix rapidyaml name, add installation"
+      patch_type: "conan"
   "0.17.0":
     - patch_file: "patches/0.17.0/0001-fix-nlohmann-include.patch"
-      base_path: "source_subfolder"
+      patch_description: "fix include path to use cci package"
+      patch_type: "conan"
     - patch_file: "patches/0.17.0/0002-cmake-fixes.patch"
-      base_path: "source_subfolder"
+      patch_description: "fix rapidyaml name, add installation"
+      patch_type: "conan"
+    - patch_file: "patches/0.17.0/0003-std-uint32_t.patch"
```

Comment:
Patches to C++ source code are a raising concern, and we are moving towards considering them a red flag in general.
If this is to force the 0.17.0 source code to be C++17 compatible, then the approach should be more like:
- Do not add the patch
- Raise a ``ConanInvalidConfiguration()`` for <=0.17.0 if cppstd>=17. 

That is a perfectly valid approach, the original 0.17.0 library didn't support 17, unless they patch it in a new 0.17.1, we should probably try to be as aligned as possible.

Reply:
@memsharded 
I agree with you.
It is very reasonable.
I will fix another PR with same issues.
Thanks a lot!